### PR TITLE
[#1] Set up the project

### LINKED ISSUE #1: Set up the project
## Why

Set up the project with [Nimble React Template](https://github.com/nimblehq/react-templates)

## Acceptance Criteria

- Have a base React application project set up
- The application should be able to run successfully

### PR DESCRIPTION
**Pull request by rosle**
close https://github.com/rosle/react-nimble-survey/issues/1

## What happened 👀

- Create the application from https://github.com/nimblehq/react-templates
- Update dependencies: there are some extra changes to fix the issues
  - Fix bootstrap issues
  - Migrate Cypress to version 10

### Bootstrap issue

There is an error

```
SassError: Undefined variable.

142 │       values: $utilities-border-colors
    │               ^^^^^^^^^^^^^^^^^^^^^^^^
```

This could be fixed by importing `bootstrap/scss/maps` which is added on Bootstrap 5.2.

Check more on Bootstrap document: https://github.com/twbs/bootstrap/blob/46bb9e78d74e0c7f4fa6f6fa611f3d9f8f28860b/site/content/docs/5.2/customize/color.md#generating-utilities

### Cypress 10 migration

There are quite lots of changes on Cypress 10. I follow the migration guide: https://docs.cypress.io/guides/references/migration-guide#Migrating-to-Cypress-version-10-0

To summarize the things I did on this PR:

- `cypress.json` -> `cypress.config.ts`
- `cypress/integration/` -> `cypress/e2e`
- Remove `plugins/index.ts`, the configuration has been moved to `cypress.config.ts`
- `support/index.ts` -> `support/e2e.ts`

The [Component Testing](https://docs.cypress.io/guides/component-testing/writing-your-first-component-test) is also in beta now. Maybe I can check more and try it out when implementing the App components.

## Proof Of Work 📹

The application runs successfully on local

Screen Shot 2565-08-03 at 14 52 35

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (hanam1ni) [inline .prettierrc:1]: The indent in this file looks odd 🤔

File: cypress/e2e/init.spec.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+describe('Cypress', () => {
+  it('is working', () => {
+    expect(true).to.equal(true);
+  });
+
+  it('visits the app', () => {
+    cy.visit('/');
```
Comment (hanam1ni) [inline cypress/e2e/init.spec.ts:7]: Let's add an extra line after this one to separate the setup and assertion part.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,92 @@
+{
```
Comment (hanam1ni) [inline package.json:1]: Let's add [engines](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#engines) to specify node and npm version.

File: src/reportWebVitals.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import type { ReportHandler } from 'web-vitals';
```
Comment (hanam1ni) [inline src/reportWebVitals.ts:1]: Just curious what is this file for? 💭

Comment (rosle) [reply src/reportWebVitals.ts:1]: This is a file generated from CRA. Seems like it is to send web metrics. 

https://create-react-app.dev/docs/measuring-performance/#sending-results-to-analytics

We can pass the ReportHandler, a function to send the metrics out. Right now we don't pass in any ReportHandler, So it is not actually being used. But the easiest way is we can pass `console.log`, and it will log the metrics out.

For example on `index.tsx`, we put `reportWebVitals(console.log);`, the metrics are logged on the console like this:

Screen Shot 2565-08-05 at 15 13 27

Right now, by default, it logs the metrics below which are CLS, FID, FCP, LCP and TTFB

But I also don't know what each metrics are for. Never play with it before 😅

File: cypress/e2e/init.spec.ts (hunk 2 above)
Comment (rosle) [reply cypress/e2e/init.spec.ts:7]: Added in [adc7753](https://github.com/rosle/react-nimble-survey/pull/9/commits/adc77533885dae70bb6ecf05fefbe95041fceff0) ✨

File: package.json (hunk 3 above)
Comment (rosle) [reply package.json:1]: Added in [adc7753](https://github.com/rosle/react-nimble-survey/pull/9/commits/adc77533885dae70bb6ecf05fefbe95041fceff0) 👍 

For this, in which case do you see the benefits, Micky? As I read, it is to provide the compatibilities range, correct? different from our `.tool-versions` which only lock the version in dev. 🤔 Wonder if it still useful for non-npm-published package like this 💭

Comment (hanam1ni) [reply package.json:1]: As I know, the only benefit is to document the compatible node and npm version in case other reviewers are not familiar with `.tool-versions` so they can check the `package.json` instead.

I noticed that I provided the wrong link to the doc 🙈 I updated the engines link in the original comment.

**Review by hanam1ni — APPROVED** [review]

File: .github/workflows/deploy.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  deploy:
+    name: Build and Deploy to Netlify
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ github.token }}
+
+      - uses: actions/checkout@v2
```
Comment (Lahphim) [inline .github/workflows/deploy.yml]:
```suggestion
      - name: Checkout
        uses: actions/checkout@v3
```
Should we use the latest one?

File: .github/workflows/deploy.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
+name: Deploy to Netlify
+
+on:
+  push:
+    branches: ['main', 'master']
```
Comment (Lahphim) [inline .github/workflows/deploy.yml]: Can we leave it just only `main`?

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup node and restore cached dependencies
+        uses: actions/setup-node@v2
```
Comment (Lahphim) [inline .github/workflows/test.yml]:
```suggestion
        uses: actions/setup-node@v3
```

File: package.json [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+  "name": "react-nimble-survey",
+  "version": "0.1.0",
+  "private": true,
+  "engines" : {
+    "npm" : ">=8.11.0",
+    "node" : ">=16.16.0"
+  },
+  "dependencies": {
```
Comment (Lahphim) [inline package.json:9]: @rosle Should we use `save-exact=true` to prevent `^` after adding new dependencies?

File: src/screens/Home.tsx [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+const HomeScreen = (): JSX.Element => {
+  const { t } = useTranslation();
+
+  return (
+    <div className="app">
+      <header className="app-header">
+        <img src={logo} className="app-logo" alt="logo" />
+        <p>{t('sample_page.message', { codeSample: '<code>src/App.tsx</code>' })}</p>
+        <a className="app-link" href="https://reactjs.org" target="_blank" rel="noopener noreferrer" data-test-id="app-link">
```
Comment (Lahphim) [inline src/screens/Home.tsx:14]: @rosle Should we move this `data-test-id` as a set of constants?
So we can export for using in the test also.

```
export const homeScreenTestIds = {
  appLink: 'app-link',
}
```

Comment (rosle) [reply src/screens/Home.tsx:14]: Yeah we should. May I do this in other components? This one would be deleted later when implementing the homepage. 🙏

File: .github/workflows/deploy.yml (hunk 6 above)
Comment (rosle) [reply .github/workflows/deploy.yml]: thank you! Fix all your comments on https://github.com/rosle/react-nimble-survey/pull/9/commits/a093c9046e886cabe7762f5e63c05779a6b56d9f ✨

**Review by Lahphim — APPROVED** [review]
